src/streamsync/core.py

```diff
@@ -1079,6 +983,10 @@ def _call_handler_callable(self, event_type, target_component, instance_path, pa
                     "headers": self.session.headers
                 }
                 arg_values.append(session_info)
+            elif arg == "ui":
+                from streamsync.ui import StreamsyncUIManager
+                ui_manager = StreamsyncUIManager(self.session.session_id)
```

Comment:
I'd pass the session tree as an argument of `StreamsyncUIManager` rather than the `session_id`.

Then you can create `StreamsyncUIManager` attached to the base tree when outside of a handler.